File: app/src/main/java/di/AppComponent.java
package di;

import com.example.kamedon.sample20.MainActivity;
import com.kamedon.kotlindagger2sample.di.AppModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by kamedon on 4/12/16.
 */
@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    void inject(MainActivity activity);
}

File: app/src/main/java/di/AppComponent.kt
package com.example.kamedon.sample20.di

import com.example.kamedon.sample20.MainActivity
import dagger.Component
import javax.inject.Singleton

/**
 * Created by kamedon on 4/12/16.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun inject(activity: MainActivity)
}
